### DISCUSSION TITLE
How to get random filtered record?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#2725] [Q&A]
Hey, I'm building a quiz app where users can participate in quizzes of different subjects. I need to somehow fetch questions randomly. I want to fetch one record at a time based on what subject the user chose. Are there any ways to accomplish that?

**Comment by ganigeorgiev** [comment]
There is a special `@random` sort key that you can use but keep in mind that the pagination will not work with it because by default we don't store the previously fetched random items:

```js
await pb.collection("example").getList(1, 20, {
    sort: "@random"
})
```

**Reply by Mr-M1M3** [reply]
We can use it with filter query param too,  right?

**Reply by ganigeorgiev** [reply]
Yes, you can combine it with the `filter` parameter.
